javascript/javascript2/week3/warm up ex/main.js

```diff
@@ -0,0 +1,168 @@
+// Warmup
+
+//1
+
+setTimeout(() => {
+  console.log("Called after 2.5 seconds");
+}, 2500);
+
+//2
+
+function logString(delay, string) {
+  setTimeout(() => {
+    console.log(string);
+  }, delay * 1000);
+}
+logString(5,"This string logged after 5 seconds")
+logString(3,"This string logged after 3 seconds")
+
+//3
+
+const ClickButton = document.querySelector("#stringLogger");
+function logOutString(){
+  logString(5, "Called after 5 seconds")
+}
+ClickButton.addEventListener("click", logOutString)
+
+//4
+
+let planetA = function earthLogger(){
+  return "Earth"
+};
+let PlanetB = function saturnLogger(){
+  return "Saturn"
+}
+function planetLogfunction(planet){
+  console.log(planet())
+}
+planetLogfunction(planetA) //Earth
+planetLogfunction(PlanetB) //Saturn
+
+//5
+
+const latitude = document.querySelector(".one");
+const longitude = document.querySelector(".two");
+const h3 = document.querySelector("h3");
+const locationButton = document.querySelector("#location");
+
+const options = {
+  enableHighAccuracy: true,
+  timeout: 5000,
+  maximumAge: 0,
+};
+
+function success(pos) {
+  const crd = pos.coords;
+
+  h3.innerHTML = "Your current position is:";
+  latitude.innerHTML = `Latitude : ${crd.latitude}`;
+  longitude.innerHTML = `Longitude: ${crd.longitude}`;
+}
+
+function error(err) {
+  console.warn(`ERROR(${err.code}): ${err.message}`);
+}
+
+function getUserLocation() {
+  navigator.geolocation.getCurrentPosition(success, error, options);
+}
+
+locationButton.addEventListener("click", getUserLocation);
+
+//7
+
+const body = document.querySelector("body");
+let callback = function chnageBgColor() {
+  body.style.backgroundColor = "red";
+  h3.style.color = "white";
+  latitude.style.color = "white";
+  longitude.style.color = "white";
+};
+
+function runAfterDelay(delay, callback) {
+  setTimeout(() => {
+    callback();
+  }, delay * 1000);
+}
+
+runAfterDelay(5, callback);
+
+//8
+
+const p = document.createElement("p");
+function doubleClick() {
+  p.innerHTML = "double click";
+  p.style.color = "white";
+  body.style.display="flex"
+  body.style.flexDirection="column"
+  body.style.alignItems="center"
+  body.appendChild(p);
+}
+
+window.addEventListener("dblclick", doubleClick);
+
+//9
+
+  function jokeCreator(shouldTellFunnyJoke, logFunnyJoke, logBadJoke ){
+    if(shouldTellFunnyJoke){
+      return logFunnyJoke()
+    } else {
+      return logBadJoke()
+    }
+  }
+
+  function logFunnyJoke(){
+    console.log("What does a storm cloud wear under his raincoat?")
+    setTimeout(() => {
+      console.log("Thunderwear")
+    }, 5000)
+  }
+
+  function logBadJoke(){
+    console.log("What do you call a can opener that doesn't work?")
+    setTimeout(() => {
+      console.log("A can't opener!")
+    }, 5000)
+  }
+
+   jokeCreator(false, logFunnyJoke, logBadJoke)
+
+// Function as a variable
+
```

Comment:
I will recommend you to store you functions like this: `
const functionsArr = [
  function func1() {
    setTimeout(() => {
      console.log("this is the first function");
    }, 2000);
  },
  function func2() {
    setTimeout(() => {
      console.log("this is the second function");
    }, 4000);
  },
  function func3() {
    setTimeout(() => {
      console.log("this is the third function");
    }, 6000);
  },
];`

and then you can call functions using loop : `// // calling all the functions using for loop
for (let i = 0; i < functionsArr.length; i++) {
  const func = functionsArr[i];
  func();
}`